client/src/app/appConfig.js

```diff
@@ -2,19 +2,9 @@
 angular.module('app').constant('appConfig', {
   // TIMESTAMP:
   // represents the name of the column with timestamp/x-data;
-  // currently such column must be present in the data, and be of ISO Date format.
-  // TODO: allow numeric or missing timestamp column ?
+  // if field timestamp is used, try parsing as data, or numeric, or fallback to iteration. 
   TIMESTAMP : "timestamp",
-  // POSSIBLE_OPF_DATA_FIELDS:
-  // Is used only in OPF files during CSV parsing, where fields may, or may not be present,
-  // depending on the user's Model settings in NuPIC.
-  // If these fields are present, we'll include them as data fields.
-  // FIXME: is this code (and guessDataFields()) needed? 'multiStepBestPredictions.5' are
-  // plotted even though not in the list.
-  POSSIBLE_OPF_DATA_FIELDS : [
-    "multiStepPredictions.actual",
-    "multiStepBestPredictions.actual"
-  ],
+  TIMESTAMP_FALLBACK : "__iter__", // no normal column should be named like this!
```

Comment:
I don't think we should assume that "__iter__" will never be a field name. I suggest we do the following instead:

* When creating the field map in generateFieldMap, if there is no `timestamp` field, we add one anyway, and set a variable, such as `useIterationForTimestamp` to `true`.
* In the `convertPapaToDygraph` function, we check the status of the `useIterationForTimestamp` variable, and add the rowId as the `timestamp` value if it is set to `true`

This seems more clean, don't you think?